core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1063,23 +1063,26 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
             case flatmap: FlatMap[Any, Any, Any, Any] =>
               updateLastTrace(flatmap.trace)
 
-              stackIndex = pushStackFrame(flatmap, stackIndex)
+              val first = flatmap.first
 
-              val result = runLoop(flatmap.first, stackIndex, stackIndex, currentDepth + 1, ops)
-              ops += 1
-
-              if (null eq result)
-                return null
+              if (first eq ZIO.unit) cur = flatmap.successK(())
+              else if (first.isInstanceOf[Success[Any]]) cur = flatmap.successK(first.asInstanceOf[Success[Any]].value)
+              else if (first.isInstanceOf[Failure[Any]]) cur = first
```

Comment:
I think let's remove these 2 branches. `Exit` overrides `flatMap` so unless we construct one somewhere manually with an exit as `first` then it won't exist. To double-check this I run the ZIO tests with the line below and it wasn't printed at all

```scala
if (flatmap.first.isInstanceOf[Success[Any]]) println("flatmap.first is Success")
```

Reply:
Done